src/App.jsx

```diff
@@ -152,8 +165,11 @@ function App() {
 
     if (currentTile.hasMine) {
       currentTile.hasExplodedMine = true;
+
+      let updatedBoard = [...currentBoard]
```

Comment:
You can avoid this clone if you follow my comments about not-mutating the prop in the functions